lucille-core/src/main/java/com/kmwllc/lucille/core/Document.java

```diff
@@ -51,6 +53,67 @@ public interface Document {
 
   void update(String name, UpdateMode mode, byte[]... values);
 
+  /**
+   * @throws IllegalArgumentException if object is not supported*/
+  public default void update(String name, UpdateMode mode, Object... values) throws Exception {
```

Comment:
All places where we throw an exception on these new methods, let's make it a DocumentException. It's ok for it to be a checked exception in this case. This is a different scenario from the other setters that take a specific type. We wanted to prevent those type-specific setters from throwing checked exceptions because we didn't want people to have to wrap code in try/catch just to add data to a Document. That's why validateFieldNames throws IllegalArgumentException, a RuntimeException. But in the case where you're setting a generic Object, it's reasonable to expect something could go wrong and to catch a checked exception.

Reply:
sounds good. What do you think about the other parts of the description? should I add default implementations for the update methods and also check to make sure that all objects to be added are of the same type?

Reply:
If you find that default update methods can be added to Document in a way that works for both implementations (JsonDocument and HashMapDocument), then yes, let's try out that refactor

yes to checking that objects are of the same type, let's try it -- I'd like to see the change

please note my other comment about avoiding reflection